tiny_dnn/core/kernels/fully_connected_op_internal.h

```diff
@@ -31,25 +31,33 @@
 namespace tiny_dnn {
 namespace kernels {
 
-inline void
-fully_connected_op_internal(const Tensor<float_t, 2>& in_data,
-                            const Tensor<float_t, 2>& weights,
-                            const Tensor<float_t, 2>& bias,
-                            Tensor<float_t, 2>*       out_data,
-                            const bool             parallelize) {
+template<typename T1, typename T2, typename T3, typename T4>
+inline void fully_connected_op_internal(const T1&  in_data,
+                                        const T2&  weights,
+                                        const T3&  bias,
+                                        T4*        out_data,
+                                        const bool parallelize) {
```

Comment:
Though shorter, I'd prefer previous style, since it's tells much more about what happening. Also previous style gives more place for optimization I believe